src/test/java/io/confluent/connect/elasticsearch/ElasticsearchClientTest.java

```diff
@@ -565,6 +566,102 @@ public void testNoVersionConflict() throws Exception {
     client2.close();
   }
 
+  @Test
+  public void testIncompatibleESVersionDataStreamNotSet() throws Exception {
+    container.close();
+    container = new ElasticsearchContainer(DEFAULT_DOCKER_IMAGE_NAME + ":" + "7.8.1");
+    container.start();
+
+    String address = container.getConnectionUrl().replace(container.getContainerIpAddress(), container.hostMachineIpAddress());
+    props.put(CONNECTION_URL_CONFIG, address);
+    props.put(CONNECTION_USERNAME_CONFIG, "elastic");
+    props.put(CONNECTION_PASSWORD_CONFIG, ELASTIC_PASSWORD);
+    config = new ElasticsearchSinkConnectorConfig(props);
+    converter = new DataConverter(config);
+
+    ElasticsearchClient client = new ElasticsearchClient(config, null);
+    helperClient = new ElasticsearchHelperClient(address, config);
```

Comment:
we can make this a separate `setupFromContainer` method that does this since its repetitive and is done in `setup()` as well